test/distributed/test_c10d.py

```diff
@@ -1711,6 +1711,37 @@ def reduce(xs, rootRank, rootTensor, op=None):
                 with self.assertRaisesRegex(RuntimeError, "Cannot use " + str(op) + " with NCCL"):
                     reduce(tensors, self.rank, rt, op)
 
+    def test_reduce_ops_stream_overwrite(self):
```

Comment:
Dedup with `test_reduce_ops` with a helper function that passes in the right options?